build/PHPStan/Build/OrChainIdenticalComparisonToInArrayRule.php

```diff
@@ -0,0 +1,163 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Build;
+
+use PhpParser\Node;
+use PhpParser\Node\Arg;
+use PhpParser\Node\ArrayItem;
+use PhpParser\Node\Expr;
+use PhpParser\Node\Expr\Array_;
+use PhpParser\Node\Expr\BinaryOp\BooleanOr;
+use PhpParser\Node\Expr\BinaryOp\Identical;
+use PhpParser\Node\Expr\ClassConstFetch;
+use PhpParser\Node\Expr\ConstFetch;
+use PhpParser\Node\Expr\FuncCall;
+use PhpParser\Node\Name;
+use PhpParser\Node\Scalar;
+use PhpParser\Node\Stmt\If_;
+use PHPStan\Analyser\Scope;
+use PHPStan\DependencyInjection\AutowiredParameter;
+use PHPStan\File\FileHelper;
+use PHPStan\Fixable\PhpPrinter;
+use PHPStan\Rules\IdentifierRuleError;
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleErrorBuilder;
+use function array_map;
+use function array_merge;
+use function array_shift;
+use function count;
+use function dirname;
+use function str_starts_with;
+
+/**
+ * @implements Rule<If_>
+ */
+final class OrChainIdenticalComparisonToInArrayRule implements Rule
+{
+
+	public function __construct(
+		#[AutowiredParameter]
+		private PhpPrinter $printer,
```

Comment:
Do not make this a DI service. Ask for ExprPrinter instead which is already a DI service.

Reply:
Thanks, I replaced it with `ExprPrinter`.